library/pkwrite.c

```diff
@@ -187,20 +211,8 @@ int mbedtls_pk_write_pubkey(unsigned char **p, unsigned char *start,
 #endif
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
     if (mbedtls_pk_get_type(key) == MBEDTLS_PK_OPAQUE) {
-        size_t buffer_size;
-
-        if (*p < start) {
-            return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
-        }
-
-        buffer_size = (size_t) (*p - start);
-        if (psa_export_public_key(key->opaque_id, start, buffer_size, &len)
-            != PSA_SUCCESS) {
-            return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
-        } else {
-            *p -= len;
-            memmove(*p, start, len);
-        }
+        mbedtls_pk_context *pk = (mbedtls_pk_context *) key;
```

Comment:
I don't like casting away `const`ness. I think `pk_write_opaque_pubkey()` should take a `const mbedtls_pk_context *` argument.

(Note: right now the EC and RSA case are not taking the full PK context as argument, but I think they will in the reshape of 7202.)